src/main/java/com/github/onsdigital/SigningKeyResolverImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.github.onsdigital;
+
+import com.github.onsdigital.exceptions.JWTDecodeException;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwsHeader;
+import io.jsonwebtoken.SigningKeyResolverAdapter;
+
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A {@link SigningKeyResolverAdapter} implementation for determining the correct, configured signing key to use based
+ * on the 'kid' field in a JWT header.
+ */
+public final class SigningKeyResolverImpl extends SigningKeyResolverAdapter {
+
+    static final String KEYS_REQUIRED_ERROR = "Public signing keys are required";
+    static final String PUBLIC_KEY_CHECK_ERROR = "Public key check failed: ";
+    static final String PUBLIC_KEY_ERROR = "Public Key not found matching 'kid'";
+    private static final String RSA = "RSA";
+
+    private final Map<String, Key> signingKeys = new HashMap<>();
```

Comment:
We are splitting class initialisation logic  between inline init and constructor init. I am being picky but I think having the construction logic all in 1 place makes the class easier to understand/read.

Can you move this new map assignment to inside the constructor with the rest of the construction logic please.